### ISSUE TITLE
Can I specify sample weights for the evaluation set while fitting a CatBoostRegressor?

### ISSUE THREAD

**Issue by gitmario** [catboost/catboost#620]
Problem: I'm trying to `fit` a `CatBoostRegressor` on imbalanced data. I would like to know if it is possible to specify a `sample_weight` parameter not only for `X` (the train set), but also for the `eval_set`
catboost version: {0.12.1.1}
Operating System: {OS X}

**Comment by annaveronika** [comment]
Yes, to do that you need to create Pool object for your eval_set

**Comment by gitmario** [comment]
Thank you Anna Veronika! I tried to implement this solution in a simple example: could you please give me feedback on this?
```
from catboost import CatBoostRegressor, Pool, FeaturesData

# X data (predictors)
x_train = pd.DataFrame({'x_numeric_1':[1,2,3,4,5,6,7],
                        'x_numeric_2':[1,3,5,1,3,5,1],
                        'x_categorical_1':["a","a","a","a","b","b","b"],
                        'x_categorical_2':["x","y","x","y","x","y","x"]})

x_eval = pd.DataFrame({'x_numeric_1':[1,2,3,4,5,6,7],
                       'x_numeric_2':[1,3,5,1,3,5,1],
                       'x_categorical_1':["a","a","a","a","b","b","b"],
                       'x_categorical_2':["x","y","x","y","x","y","x"]})

numeric_columns = ['x_numeric_1', 'x_numeric_2']
categorical_columns = ['x_categorical_1', 'x_categorical_1']
cat_features = [2,3]

# y data (target)
y_train = np.array([1,3,1,4,1,5,1])
y_eval  = np.array([1,4,1,3,1,6,1])

# w data (observation weights)
w_train = np.array([0.1,0.1,0.1,0.3,0.3,0.3,0.5])
w_eval  = np.array([0.2,0.2,0.1,0.4,0.3,0.2,0.1])

# Initialize CatBoostRegressor
model = CatBoostRegressor(iterations=2, learning_rate=1, depth=2)

data_eval_weighted = Pool(
    data=FeaturesData(
        num_feature_data=np.array(
            x_eval[numeric_columns].values,
            dtype=np.float32
        ),
        cat_feature_data=np.array(
            x_eval[categorical_columns].values,
            dtype=object
        )
    ),
    label=y_eval,
    weight=w_eval
)

# Fit model
model.fit(X=x_train,
          y=y_train,
          sample_weight=w_train,
          eval_set=data_eval_weighted,
          cat_features=cat_features)
```

**Comment by gitmario** [comment]
Thank you again. If I wanted to add an evaluation set with observation weights, should I modify your example like the following?

```
from catboost import CatBoostClassifier, Pool

train_data = Pool(
    data=[[1, 4, 5, 6], 
          [4, 5, 6, 7], 
          [30, 40, 50, 60]],
    label=[1, 1, -1],
    weight=[0.1, 0.2, 0.3]
)

eval_data = Pool(
    data=[[1, 4, 5, 6], 
          [4, 5, 6, 7], 
          [30, 40, 50, 60]],
    label=[1, 0, -1],
    weight=[0.7, 0.1, 0.3]
)

model = CatBoostClassifier(iterations = 10)

model.fit(X=train_data, eval_set=eval_data)
preds_class = model.predict(train_data)
```

**Comment by annaveronika** [comment]
Exactly!

**Comment by annaveronika** [comment]
You can also use dataframes or numpy arrays, this is just the simplest example

**Comment by gitmario** [comment]
Awesome

**Comment by gitmario** [comment]
By the way, is the weight implemented with under/oversampling, or by modifying the cost function?

**Comment by annaveronika** [comment]
By modifying the cost function and also the metrics. You can see here https://tech.yandex.com/catboost/doc/dg/concepts/loss-functions-docpage/ how they are using the weights.

**Comment by gitmario** [comment]
Great, thanks. Issue resolved

**Comment by annaveronika** [comment]
Awesome! We will also add Q&A section into our docs, and we'll add your question there.
